### PR TITLE
X-Pages Example 

### PR DESCRIPTION
**Pull request by GoldenGnu**
Added X-Pages example to MarketApiTest

### CODE REVIEW

File: src/main/java/net/troja/eve/esi/HeaderUtil.java [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+     */
+    public static Date getExpires(Map<String, List<String>> responseHeaders) {
+        //Get header
+        String header = getHeader(responseHeaders, "Expires");
+        if (header == null) {
+            return null;
+        }
+
+        //Convert
+        Date xPages = null;
```
Comment (deadlybulb) [inline src/main/java/net/troja/eve/esi/HeaderUtil.java]: Change var name to "expires"?

File: src/main/java/net/troja/eve/esi/HeaderUtil.java [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    private HeaderUtil() { }
+
+    /**
+     * 
+     * @param responseHeaders ApiClient.getResponseHeaders()
+     * @return 
+     */
+    public static Integer getXPages(Map<String, List<String>> responseHeaders) {
+        //Get header
+        String header = getHeader(responseHeaders, "X-Pages");
```
Comment (deadlybulb) [inline src/main/java/net/troja/eve/esi/HeaderUtil.java]: If we want to keep this code long term, suggest you make "X-Pages" and "Expires" into final static constants.

File: src/main/java/net/troja/eve/esi/HeaderUtil.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+
+import com.fasterxml.jackson.databind.util.ISO8601DateFormat;
+import java.text.ParseException;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+public class HeaderUtil {
+
+    private static final ISO8601DateFormat DATE_FORMAT = new ISO8601DateFormat();
```
Comment (deadlybulb) [inline src/main/java/net/troja/eve/esi/HeaderUtil.java]: This one is thread-safe, right?

File: src/main/java/net/troja/eve/esi/HeaderUtil.java [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+     * @param header
+     * @return 
+     */
+    public static String getHeader(Map<String, List<String>> responseHeaders, String header) {
+        if (responseHeaders == null) {
+            return null;
+        }
+        //Search the headers case insensitive (headers should be evaluated as case insensitive, but, the Swagger implementation uses HashMap that is case sensitive)
+        List<String> headersList = null;
+        for (Map.Entry<String, List<String>> entry : responseHeaders.entrySet()) {
```
Comment (deadlybulb) [inline src/main/java/net/troja/eve/esi/HeaderUtil.java]: Based on your logic below, headerList doesn't need to be a list as you only ever retrieve the first value.  You can make this a String instead, and your assignment logic below can go from:

headersList = entry.getValue();

to

headersList = entry.getValue().isEmpty() ? null : entry.getValue().get(0)

you can clean that up slightly if you don't want to call getValue twice.

File: src/main/java/net/troja/eve/esi/HeaderUtil.java (hunk 3 above)
Comment (GoldenGnu) [reply src/main/java/net/troja/eve/esi/HeaderUtil.java]: It say it is in documentation
> Provide a fast thread-safe formatter/parser DateFormat for ISO8601 dates ONLY.
It was mainly done to be used with Jackson JSON Processor.

File: src/main/java/net/troja/eve/esi/HeaderUtil.java (hunk 2 above)
Comment (GoldenGnu) [reply src/main/java/net/troja/eve/esi/HeaderUtil.java]: Good suggestion, will update the PR

File: src/main/java/net/troja/eve/esi/HeaderUtil.java (hunk 1 above)
Comment (GoldenGnu) [reply src/main/java/net/troja/eve/esi/HeaderUtil.java]: Yes, ofc. :)

**Pull request merged**